### ISSUE TITLE
[Bug] CLI should be more verbose about errors from the VM

### ISSUE THREAD

**Issue by bob-wilson** [libra/libra#3663]
# 🐛 Bug

This is a usability bug for the Libra CLI. When there is an error from the VM, it is way too easy to miss it because the only indication that a transaction fails is that no events are emitted. If at all possible, we should find a way to show some error messages.

## To reproduce

Follow the steps in the [Run Move Programs Locally document](https://developers.libra.org/docs/run-move-locally) up through the "Execute Transaction Script" section. Re-run the custom script without specifying any script arguments (or with completely bogus arguments like "abc"). You should see the following output:

```
transaction is stored!
no events emitted
Successfully finished execution
```

The only difference from the successful output is the "no events emitted". It is especially confusing because of the "Successfully finished execution" message.

## Expected Behavior

Whatever error is detected by the VM should be displayed in a more prominent way.

**Comment by bob-wilson** [comment]
See also https://github.com/libra/libra/issues/3658 -- the problem is even worse for failures publishing modules because (AFAICS) there is no way at all to see if the operation was successful. The CLI shows "no events emitted" and "Successfully published module" regardless.

**Comment by bob-wilson** [comment]
This is fixed in #4017
